# Profitable App Profiles for the App Store and Google Play Markets

- We only build free apps and our revenue consists of in-app ads. Therefore, our revenue for any given app is mostly influenced by the number of users.
- Goal for this project is to analyze data to help our developers understand what type of apps are likely to attract more users.

### Dataset
- A data set containing data about approximately 10,000 Android apps from Google Play; the data was collected in August 2018. 
- A data set containing data about approximately 7,000 iOS apps from the App Store; the data was collected in July 2017. 

### Opening and Exploring Datasets



In [26]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

        
# dataset: list of lists
# start, end: integers, staring and the ending indices of a slice from dataset


In [27]:
file = open('/Users/kitaeklee/Desktop/Data/AppStore/AppleStore.csv')

from csv import reader
read_file = reader(file)
data_list = list(read_file)
data_list


[['',
  'id',
  'track_name',
  'size_bytes',
  'currency',
  'price',
  'rating_count_tot',
  'rating_count_ver',
  'user_rating',
  'user_rating_ver',
  'ver',
  'cont_rating',
  'prime_genre',
  'sup_devices.num',
  'ipadSc_urls.num',
  'lang.num',
  'vpp_lic'],
 ['1',
  '281656475',
  'PAC-MAN Premium',
  '100788224',
  'USD',
  '3.99',
  '21292',
  '26',
  '4',
  '4.5',
  '6.3.5',
  '4+',
  'Games',
  '38',
  '5',
  '10',
  '1'],
 ['2',
  '281796108',
  'Evernote - stay organized',
  '158578688',
  'USD',
  '0',
  '161065',
  '26',
  '4',
  '3.5',
  '8.2.2',
  '4+',
  'Productivity',
  '37',
  '5',
  '23',
  '1'],
 ['3',
  '281940292',
  'WeatherBug - Local Weather, Radar, Maps, Alerts',
  '100524032',
  'USD',
  '0',
  '188583',
  '2822',
  '3.5',
  '4.5',
  '5.0.0',
  '4+',
  'Weather',
  '37',
  '5',
  '3',
  '1'],
 ['4',
  '282614216',
  'eBay: Best App to Buy, Sell, Save! Online Shopping',
  '128512000',
  'USD',
  '0',
  '262241',
  '649',
  '4',
  '4.5',
  '5.10.0',
  '12+'

In [28]:
file2 = open('/Users/kitaeklee/Desktop/Data/AppStore/googleplaystore.csv')

from csv import reader
read_file2 = reader(file2)
data_list2 = list(read_file2)
data_list2

[['App',
  'Category',
  'Rating',
  'Reviews',
  'Size',
  'Installs',
  'Type',
  'Price',
  'Content Rating',
  'Genres',
  'Last Updated',
  'Current Ver',
  'Android Ver'],
 ['Photo Editor & Candy Camera & Grid & ScrapBook',
  'ART_AND_DESIGN',
  '4.1',
  '159',
  '19M',
  '10,000+',
  'Free',
  '0',
  'Everyone',
  'Art & Design',
  'January 7, 2018',
  '1.0.0',
  '4.0.3 and up'],
 ['Coloring book moana',
  'ART_AND_DESIGN',
  '3.9',
  '967',
  '14M',
  '500,000+',
  'Free',
  '0',
  'Everyone',
  'Art & Design;Pretend Play',
  'January 15, 2018',
  '2.0.0',
  '4.0.3 and up'],
 ['U Launcher Lite – FREE Live Cool Themes, Hide Apps',
  'ART_AND_DESIGN',
  '4.7',
  '87510',
  '8.7M',
  '5,000,000+',
  'Free',
  '0',
  'Everyone',
  'Art & Design',
  'August 1, 2018',
  '1.2.4',
  '4.0.3 and up'],
 ['Sketch - Draw & Paint',
  'ART_AND_DESIGN',
  '4.5',
  '215644',
  '25M',
  '50,000,000+',
  'Free',
  '0',
  'Teen',
  'Art & Design',
  'June 8, 2018',
  'Varies with device',
  '4.2 a

In [29]:
explore_data(data_list, 1, 3)

['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']




In [30]:
explore_data(data_list2, 1, 3)

['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']




In [31]:
# column names of first dataset
explore_data(data_list, 0, 1)

['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']




In [32]:
# column names of second dataset
explore_data(data_list2, 0, 1)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']




# Data Cleaning

##### Apple Store Dataset

In [33]:
import pandas as pd

# Create dataframe using list of lists of the first dataset
df = pd.DataFrame(data_list)
header = ['','id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']
df = df[1:]
df.columns = header
df = df.drop(df.columns[0], axis=1)

df

id                                         track_name  \
1      281656475                                    PAC-MAN Premium   
2      281796108                          Evernote - stay organized   
3      281940292    WeatherBug - Local Weather, Radar, Maps, Alerts   
4      282614216  eBay: Best App to Buy, Sell, Save! Online Shop...   
5      282935706                                              Bible   
...          ...                                                ...   
7193  1187617475                                              Kubik   
7194  1187682390                                  VR Roller-Coaster   
7195  1187779532              Bret Michaels Emojis + Lyric Keyboard   
7196  1187838770          VR Roller Coaster World - Virtual Reality   
7197  1188375727                       Escape the Sweet Shop Series   

     size_bytes currency price rating_count_tot rating_count_ver user_rating  \
1     100788224      USD  3.99            21292               26           4   
2     158578688      USD     0           161065               26           4   
3     100524032      USD     0           188583             2822         3.5   
4     128512000      USD     0           262241              649           4   
5      92774400      USD     0           985920             5320         4.5   
...         ...      ...   ...              ...              ...         ...   
7193  126644224      USD     0              142               75         4.5   
7194  120760320      USD     0               30               30         4.5   
7195  111322112      USD  1.99               15                0         4.5   
7196   97235968      USD     0               85               32         4.5   
7197   90898432      USD     0                3                3           5   

     user_rating_ver     ver cont_rating   prime_genre sup_devices.num  \
1                4.5   6.3.5          4+         Games              38   
2                3.5   8.2.2          4+  Productivity              37   
3                4.5   5.0.0          4+       Weather              37   
4                4.5  5.10.0         12+      Shopping              37   
5                  5   7.5.1          4+     Reference              37   
...              ...     ...         ...           ...             ...   
7193             4.5     1.3          4+         Games              38   
7194             4.5     0.9          4+         Games              38   
7195               0   1.0.2          9+     Utilities              37   
7196             4.5  1.0.15         12+         Games              38   
7197               5     1.0          4+         Games              40   

     ipadSc_urls.num lang.num vpp_lic  
1                  5       10       1  
2                  5       23       1  
3                  5        3       1  
4                  5        9       1  
5                  5       45       1  
...              ...      ...     ...  
7193               5        1       1  
7194               0        1       1  
7195               1        1       1  
7196               0        2       1  
7197               0        2       1  

[7197 rows x 16 columns]

Check the number of duplicates that have the same name of applications.

In [34]:
# Number of duplicates in the AppStore dataset
duplicate_list=[]
for row in df.duplicated(subset='track_name'):
    if row == True:
        duplicate_list.append(row)
num = len(duplicate_list)
print("Number of duplicates:", num)


Number of duplicates: 2


There are 2 duplicates therefore, remove the duplicates based on the name of the application but keep the first one.

In [35]:
# Drop duplicates
df.drop_duplicates(subset='track_name', keep='first', inplace=True)

df

id                                         track_name  \
1      281656475                                    PAC-MAN Premium   
2      281796108                          Evernote - stay organized   
3      281940292    WeatherBug - Local Weather, Radar, Maps, Alerts   
4      282614216  eBay: Best App to Buy, Sell, Save! Online Shop...   
5      282935706                                              Bible   
...          ...                                                ...   
7193  1187617475                                              Kubik   
7194  1187682390                                  VR Roller-Coaster   
7195  1187779532              Bret Michaels Emojis + Lyric Keyboard   
7196  1187838770          VR Roller Coaster World - Virtual Reality   
7197  1188375727                       Escape the Sweet Shop Series   

     size_bytes currency price rating_count_tot rating_count_ver user_rating  \
1     100788224      USD  3.99            21292               26           4   
2     158578688      USD     0           161065               26           4   
3     100524032      USD     0           188583             2822         3.5   
4     128512000      USD     0           262241              649           4   
5      92774400      USD     0           985920             5320         4.5   
...         ...      ...   ...              ...              ...         ...   
7193  126644224      USD     0              142               75         4.5   
7194  120760320      USD     0               30               30         4.5   
7195  111322112      USD  1.99               15                0         4.5   
7196   97235968      USD     0               85               32         4.5   
7197   90898432      USD     0                3                3           5   

     user_rating_ver     ver cont_rating   prime_genre sup_devices.num  \
1                4.5   6.3.5          4+         Games              38   
2                3.5   8.2.2          4+  Productivity              37   
3                4.5   5.0.0          4+       Weather              37   
4                4.5  5.10.0         12+      Shopping              37   
5                  5   7.5.1          4+     Reference              37   
...              ...     ...         ...           ...             ...   
7193             4.5     1.3          4+         Games              38   
7194             4.5     0.9          4+         Games              38   
7195               0   1.0.2          9+     Utilities              37   
7196             4.5  1.0.15         12+         Games              38   
7197               5     1.0          4+         Games              40   

     ipadSc_urls.num lang.num vpp_lic  
1                  5       10       1  
2                  5       23       1  
3                  5        3       1  
4                  5        9       1  
5                  5       45       1  
...              ...      ...     ...  
7193               5        1       1  
7194               0        1       1  
7195               1        1       1  
7196               0        2       1  
7197               0        2       1  

[7195 rows x 16 columns]

Our firm only build apps that are free to download and install. Therefore, recreate the dataframe only with the apps that are free.

In [36]:
# Drop the rows with non-free apps since we only deal with free apps.
df = df[df['price'] == '0']

In [37]:
# Drop rows with missing values just in case
df.dropna(axis=0, inplace=True)

/Users/kitaeklee/opt/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  


In [38]:
# Reset the index
df.reset_index(drop=True, inplace=True)
df

id                                         track_name  \
0      281796108                          Evernote - stay organized   
1      281940292    WeatherBug - Local Weather, Radar, Maps, Alerts   
2      282614216  eBay: Best App to Buy, Sell, Save! Online Shop...   
3      282935706                                              Bible   
4      283646709             PayPal - Send and request money safely   
...          ...                                                ...   
4049  1186384912       Demolition Derby Virtual Reality (VR) Racing   
4050  1187617475                                              Kubik   
4051  1187682390                                  VR Roller-Coaster   
4052  1187838770          VR Roller Coaster World - Virtual Reality   
4053  1188375727                       Escape the Sweet Shop Series   

     size_bytes currency price rating_count_tot rating_count_ver user_rating  \
0     158578688      USD     0           161065               26           4   
1     100524032      USD     0           188583             2822         3.5   
2     128512000      USD     0           262241              649           4   
3      92774400      USD     0           985920             5320         4.5   
4     227795968      USD     0           119487              879           4   
...         ...      ...   ...              ...              ...         ...   
4049  168774656      USD     0               18               18           4   
4050  126644224      USD     0              142               75         4.5   
4051  120760320      USD     0               30               30         4.5   
4052   97235968      USD     0               85               32         4.5   
4053   90898432      USD     0                3                3           5   

     user_rating_ver     ver cont_rating   prime_genre sup_devices.num  \
0                3.5   8.2.2          4+  Productivity              37   
1                4.5   5.0.0          4+       Weather              37   
2                4.5  5.10.0         12+      Shopping              37   
3                  5   7.5.1          4+     Reference              37   
4                4.5  6.12.0          4+       Finance              37   
...              ...     ...         ...           ...             ...   
4049               4   1.0.0         12+         Games              38   
4050             4.5     1.3          4+         Games              38   
4051             4.5     0.9          4+         Games              38   
4052             4.5  1.0.15         12+         Games              38   
4053               5     1.0          4+         Games              40   

     ipadSc_urls.num lang.num vpp_lic  
0                  5       23       1  
1                  5        3       1  
2                  5        9       1  
3                  5       45       1  
4                  0       19       1  
...              ...      ...     ...  
4049               4        1       1  
4050               5        1       1  
4051               0        1       1  
4052               0        2       1  
4053               0        2       1  

[4054 rows x 16 columns]

##### Google Playstore Dataset

In [39]:
# Create a dataframe using the list of lists of the second dataset
df2 = pd.DataFrame(data_list2)
df2 = df2[1:]
header2 = ['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']
df2.columns = header2


df2

App             Category  \
1         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
2                                    Coloring book moana       ART_AND_DESIGN   
3      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
4                                  Sketch - Draw & Paint       ART_AND_DESIGN   
5                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
...                                                  ...                  ...   
10837                                   Sya9a Maroc - FR               FAMILY   
10838                   Fr. Mike Schmitz Audio Teachings               FAMILY   
10839                             Parkinson Exercices FR              MEDICAL   
10840                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10841      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

      Rating Reviews                Size     Installs  Type Price  \
1        4.1     159                 19M      10,000+  Free     0   
2        3.9     967                 14M     500,000+  Free     0   
3        4.7   87510                8.7M   5,000,000+  Free     0   
4        4.5  215644                 25M  50,000,000+  Free     0   
5        4.3     967                2.8M     100,000+  Free     0   
...      ...     ...                 ...          ...   ...   ...   
10837    4.5      38                 53M       5,000+  Free     0   
10838    5.0       4                3.6M         100+  Free     0   
10839    NaN       3                9.5M       1,000+  Free     0   
10840    4.5     114  Varies with device       1,000+  Free     0   
10841    4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating                     Genres      Last Updated  \
1           Everyone               Art & Design   January 7, 2018   
2           Everyone  Art & Design;Pretend Play  January 15, 2018   
3           Everyone               Art & Design    August 1, 2018   
4               Teen               Art & Design      June 8, 2018   
5           Everyone    Art & Design;Creativity     June 20, 2018   
...              ...                        ...               ...   
10837       Everyone                  Education     July 25, 2017   
10838       Everyone                  Education      July 6, 2018   
10839       Everyone                    Medical  January 20, 2017   
10840     Mature 17+          Books & Reference  January 19, 2015   
10841       Everyone                  Lifestyle     July 25, 2018   

              Current Ver         Android Ver  
1                   1.0.0        4.0.3 and up  
2                   2.0.0        4.0.3 and up  
3                   1.2.4        4.0.3 and up  
4      Varies with device          4.2 and up  
5                     1.1          4.4 and up  
...                   ...                 ...  
10837                1.48          4.1 and up  
10838                 1.0          4.1 and up  
10839                 1.0          2.2 and up  
10840  Varies with device  Varies with device  
10841  Varies with device  Varies with device  

[10841 rows x 13 columns]

Based on discussion on Kaggle, there is missing value in one row.

In [40]:
# Based on discussion on Kaggle, there is missing value.
df2.loc[10473, :]

App               Life Made WI-Fi Touchscreen Photo Frame
Category                                              1.9
Rating                                                 19
Reviews                                              3.0M
Size                                               1,000+
Installs                                             Free
Type                                                    0
Price                                            Everyone
Content Rating                                           
Genres                                  February 11, 2018
Last Updated                                       1.0.19
Current Ver                                    4.0 and up
Android Ver                                          None
Name: 10473, dtype: object

There is missing value in Content Rating. Therefore, drop that row.

In [41]:
# Drop rows with missing values
df2.dropna(axis=0, inplace=True)
df2

App             Category  \
1         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
2                                    Coloring book moana       ART_AND_DESIGN   
3      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
4                                  Sketch - Draw & Paint       ART_AND_DESIGN   
5                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
...                                                  ...                  ...   
10837                                   Sya9a Maroc - FR               FAMILY   
10838                   Fr. Mike Schmitz Audio Teachings               FAMILY   
10839                             Parkinson Exercices FR              MEDICAL   
10840                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10841      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

      Rating Reviews                Size     Installs  Type Price  \
1        4.1     159                 19M      10,000+  Free     0   
2        3.9     967                 14M     500,000+  Free     0   
3        4.7   87510                8.7M   5,000,000+  Free     0   
4        4.5  215644                 25M  50,000,000+  Free     0   
5        4.3     967                2.8M     100,000+  Free     0   
...      ...     ...                 ...          ...   ...   ...   
10837    4.5      38                 53M       5,000+  Free     0   
10838    5.0       4                3.6M         100+  Free     0   
10839    NaN       3                9.5M       1,000+  Free     0   
10840    4.5     114  Varies with device       1,000+  Free     0   
10841    4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating                     Genres      Last Updated  \
1           Everyone               Art & Design   January 7, 2018   
2           Everyone  Art & Design;Pretend Play  January 15, 2018   
3           Everyone               Art & Design    August 1, 2018   
4               Teen               Art & Design      June 8, 2018   
5           Everyone    Art & Design;Creativity     June 20, 2018   
...              ...                        ...               ...   
10837       Everyone                  Education     July 25, 2017   
10838       Everyone                  Education      July 6, 2018   
10839       Everyone                    Medical  January 20, 2017   
10840     Mature 17+          Books & Reference  January 19, 2015   
10841       Everyone                  Lifestyle     July 25, 2018   

              Current Ver         Android Ver  
1                   1.0.0        4.0.3 and up  
2                   2.0.0        4.0.3 and up  
3                   1.2.4        4.0.3 and up  
4      Varies with device          4.2 and up  
5                     1.1          4.4 and up  
...                   ...                 ...  
10837                1.48          4.1 and up  
10838                 1.0          4.1 and up  
10839                 1.0          2.2 and up  
10840  Varies with device  Varies with device  
10841  Varies with device  Varies with device  

[10840 rows x 13 columns]

Check the number of duplicates that have the same name of the application.

In [42]:
# Number of duplicates in the Googlestore dataset
duplicate_list2=[]
for row in df2.duplicated(subset='App'):
    if row == True:
        duplicate_list2.append(row)
num2 = len(duplicate_list2)
print("Number of duplicates:", num2)

Number of duplicates: 1181


There are 1181 duplicates. Drop the duplicates based on the name of the applications while keeping the first duplicate.

In [43]:
# Drop duplicates
df2.drop_duplicates(subset='App', keep='first', inplace=True)

df2

App             Category  \
1         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
2                                    Coloring book moana       ART_AND_DESIGN   
3      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
4                                  Sketch - Draw & Paint       ART_AND_DESIGN   
5                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
...                                                  ...                  ...   
10837                                   Sya9a Maroc - FR               FAMILY   
10838                   Fr. Mike Schmitz Audio Teachings               FAMILY   
10839                             Parkinson Exercices FR              MEDICAL   
10840                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10841      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

      Rating Reviews                Size     Installs  Type Price  \
1        4.1     159                 19M      10,000+  Free     0   
2        3.9     967                 14M     500,000+  Free     0   
3        4.7   87510                8.7M   5,000,000+  Free     0   
4        4.5  215644                 25M  50,000,000+  Free     0   
5        4.3     967                2.8M     100,000+  Free     0   
...      ...     ...                 ...          ...   ...   ...   
10837    4.5      38                 53M       5,000+  Free     0   
10838    5.0       4                3.6M         100+  Free     0   
10839    NaN       3                9.5M       1,000+  Free     0   
10840    4.5     114  Varies with device       1,000+  Free     0   
10841    4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating                     Genres      Last Updated  \
1           Everyone               Art & Design   January 7, 2018   
2           Everyone  Art & Design;Pretend Play  January 15, 2018   
3           Everyone               Art & Design    August 1, 2018   
4               Teen               Art & Design      June 8, 2018   
5           Everyone    Art & Design;Creativity     June 20, 2018   
...              ...                        ...               ...   
10837       Everyone                  Education     July 25, 2017   
10838       Everyone                  Education      July 6, 2018   
10839       Everyone                    Medical  January 20, 2017   
10840     Mature 17+          Books & Reference  January 19, 2015   
10841       Everyone                  Lifestyle     July 25, 2018   

              Current Ver         Android Ver  
1                   1.0.0        4.0.3 and up  
2                   2.0.0        4.0.3 and up  
3                   1.2.4        4.0.3 and up  
4      Varies with device          4.2 and up  
5                     1.1          4.4 and up  
...                   ...                 ...  
10837                1.48          4.1 and up  
10838                 1.0          4.1 and up  
10839                 1.0          2.2 and up  
10840  Varies with device  Varies with device  
10841  Varies with device  Varies with device  

[9659 rows x 13 columns]

Our firm only build apps that are free to download and install. Therefore, recreate the dataframe only with the apps that are free.

In [44]:
# Drop the rows with non-free apps since we only deal with free apps.
df2 = df2[df2['Price']=='0']

In [45]:
# Reset the index
df2.reset_index(drop=True, inplace=True)
df2

App             Category  \
0        Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                   Coloring book moana       ART_AND_DESIGN   
2     U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                                 Sketch - Draw & Paint       ART_AND_DESIGN   
4                 Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
...                                                 ...                  ...   
8898                                   Sya9a Maroc - FR               FAMILY   
8899                   Fr. Mike Schmitz Audio Teachings               FAMILY   
8900                             Parkinson Exercices FR              MEDICAL   
8901                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
8902      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

     Rating Reviews                Size     Installs  Type Price  \
0       4.1     159                 19M      10,000+  Free     0   
1       3.9     967                 14M     500,000+  Free     0   
2       4.7   87510                8.7M   5,000,000+  Free     0   
3       4.5  215644                 25M  50,000,000+  Free     0   
4       4.3     967                2.8M     100,000+  Free     0   
...     ...     ...                 ...          ...   ...   ...   
8898    4.5      38                 53M       5,000+  Free     0   
8899    5.0       4                3.6M         100+  Free     0   
8900    NaN       3                9.5M       1,000+  Free     0   
8901    4.5     114  Varies with device       1,000+  Free     0   
8902    4.5  398307                 19M  10,000,000+  Free     0   

     Content Rating                     Genres      Last Updated  \
0          Everyone               Art & Design   January 7, 2018   
1          Everyone  Art & Design;Pretend Play  January 15, 2018   
2          Everyone               Art & Design    August 1, 2018   
3              Teen               Art & Design      June 8, 2018   
4          Everyone    Art & Design;Creativity     June 20, 2018   
...             ...                        ...               ...   
8898       Everyone                  Education     July 25, 2017   
8899       Everyone                  Education      July 6, 2018   
8900       Everyone                    Medical  January 20, 2017   
8901     Mature 17+          Books & Reference  January 19, 2015   
8902       Everyone                  Lifestyle     July 25, 2018   

             Current Ver         Android Ver  
0                  1.0.0        4.0.3 and up  
1                  2.0.0        4.0.3 and up  
2                  1.2.4        4.0.3 and up  
3     Varies with device          4.2 and up  
4                    1.1          4.4 and up  
...                  ...                 ...  
8898                1.48          4.1 and up  
8899                 1.0          4.1 and up  
8900                 1.0          2.2 and up  
8901  Varies with device  Varies with device  
8902  Varies with device  Varies with device  

[8903 rows x 13 columns]

### Removing non-English applications

Since our firm only use English for the apps we develop, and we'd like to analyze only the apps that are directed toward an English-speaking user, we have to remove the non-English based applications. 

We would like to remove each app with a name containing a symbol that is not commonly used in English text — English text usually includes letters from the English alphabet, numbers composed of digits from 0 to 9, punctuation marks (., !, ?, ;), and other symbols (+, *, /).

Behind the scenes, each character we use in a string has a corresponding number associated with it.

We can get the corresponding number of each character using the ord().

The numbers corresponding to the characters we commonly use in an English text are all in the range 0 to 127, according to the ASCII (American Standard Code for Information Interchange) system. Based on this number range, we can build a function that detects whether a character belongs to the set of common English characters or not. If the number is equal to or less than 127, then the character belongs to the set of common English characters.

Use indexing to select an individual character from strings, and we can also iterate on the string using a for loop.

However, there are some strings that includes emojis while rest of the letters are English. Therfore to minimize the data loss, only drop the rows that contains more than 3 non-English letter.

Define a function that iterate over given strings and return True if more than 3 letters in the string are non-English.

In [46]:
def iseng(string):
    num = 0
    for i in range(len(string)):
        if ord(string[i]) > 127:
            num += 1
        if num > 3:
            return True


Drop the rows with more than 3 non-English letters in the name of applications from Apple Store Dataset.

In [47]:
for row in df.itertuples():
    if iseng(str(row[2])) == True:
        idx = row[0]       
        df.drop(index=idx, inplace=True, axis=0)
df.reset_index(inplace=True, drop=True)

In [48]:
df

id                                         track_name  \
0      281796108                          Evernote - stay organized   
1      281940292    WeatherBug - Local Weather, Radar, Maps, Alerts   
2      282614216  eBay: Best App to Buy, Sell, Save! Online Shop...   
3      282935706                                              Bible   
4      283646709             PayPal - Send and request money safely   
...          ...                                                ...   
3215  1186384912       Demolition Derby Virtual Reality (VR) Racing   
3216  1187617475                                              Kubik   
3217  1187682390                                  VR Roller-Coaster   
3218  1187838770          VR Roller Coaster World - Virtual Reality   
3219  1188375727                       Escape the Sweet Shop Series   

     size_bytes currency price rating_count_tot rating_count_ver user_rating  \
0     158578688      USD     0           161065               26           4   
1     100524032      USD     0           188583             2822         3.5   
2     128512000      USD     0           262241              649           4   
3      92774400      USD     0           985920             5320         4.5   
4     227795968      USD     0           119487              879           4   
...         ...      ...   ...              ...              ...         ...   
3215  168774656      USD     0               18               18           4   
3216  126644224      USD     0              142               75         4.5   
3217  120760320      USD     0               30               30         4.5   
3218   97235968      USD     0               85               32         4.5   
3219   90898432      USD     0                3                3           5   

     user_rating_ver     ver cont_rating   prime_genre sup_devices.num  \
0                3.5   8.2.2          4+  Productivity              37   
1                4.5   5.0.0          4+       Weather              37   
2                4.5  5.10.0         12+      Shopping              37   
3                  5   7.5.1          4+     Reference              37   
4                4.5  6.12.0          4+       Finance              37   
...              ...     ...         ...           ...             ...   
3215               4   1.0.0         12+         Games              38   
3216             4.5     1.3          4+         Games              38   
3217             4.5     0.9          4+         Games              38   
3218             4.5  1.0.15         12+         Games              38   
3219               5     1.0          4+         Games              40   

     ipadSc_urls.num lang.num vpp_lic  
0                  5       23       1  
1                  5        3       1  
2                  5        9       1  
3                  5       45       1  
4                  0       19       1  
...              ...      ...     ...  
3215               4        1       1  
3216               5        1       1  
3217               0        1       1  
3218               0        2       1  
3219               0        2       1  

[3220 rows x 16 columns]

Drop the rows with more than 3 non-English letters in the name of applications from Google Playstore Dataset.

In [49]:
for row in df2.itertuples():
    if iseng(str(row[1])) == True:
        idx = row[0]       
        df2.drop(index=idx, inplace=True, axis=0)
df2.reset_index(inplace=True, drop=True)

In [50]:
df2

App             Category  \
0        Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                   Coloring book moana       ART_AND_DESIGN   
2     U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                                 Sketch - Draw & Paint       ART_AND_DESIGN   
4                 Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
...                                                 ...                  ...   
8857                                   Sya9a Maroc - FR               FAMILY   
8858                   Fr. Mike Schmitz Audio Teachings               FAMILY   
8859                             Parkinson Exercices FR              MEDICAL   
8860                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
8861      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

     Rating Reviews                Size     Installs  Type Price  \
0       4.1     159                 19M      10,000+  Free     0   
1       3.9     967                 14M     500,000+  Free     0   
2       4.7   87510                8.7M   5,000,000+  Free     0   
3       4.5  215644                 25M  50,000,000+  Free     0   
4       4.3     967                2.8M     100,000+  Free     0   
...     ...     ...                 ...          ...   ...   ...   
8857    4.5      38                 53M       5,000+  Free     0   
8858    5.0       4                3.6M         100+  Free     0   
8859    NaN       3                9.5M       1,000+  Free     0   
8860    4.5     114  Varies with device       1,000+  Free     0   
8861    4.5  398307                 19M  10,000,000+  Free     0   

     Content Rating                     Genres      Last Updated  \
0          Everyone               Art & Design   January 7, 2018   
1          Everyone  Art & Design;Pretend Play  January 15, 2018   
2          Everyone               Art & Design    August 1, 2018   
3              Teen               Art & Design      June 8, 2018   
4          Everyone    Art & Design;Creativity     June 20, 2018   
...             ...                        ...               ...   
8857       Everyone                  Education     July 25, 2017   
8858       Everyone                  Education      July 6, 2018   
8859       Everyone                    Medical  January 20, 2017   
8860     Mature 17+          Books & Reference  January 19, 2015   
8861       Everyone                  Lifestyle     July 25, 2018   

             Current Ver         Android Ver  
0                  1.0.0        4.0.3 and up  
1                  2.0.0        4.0.3 and up  
2                  1.2.4        4.0.3 and up  
3     Varies with device          4.2 and up  
4                    1.1          4.4 and up  
...                  ...                 ...  
8857                1.48          4.1 and up  
8858                 1.0          4.1 and up  
8859                 1.0          2.2 and up  
8860  Varies with device  Varies with device  
8861  Varies with device  Varies with device  

[8862 rows x 13 columns]

Check if Applications like the ones below are dropped from the dataset.
- 'Docs To Go™ Free Office Suite'
- 'Instachat 😜'
- '爱奇艺PPS -《欢乐颂2》电视剧热播'


In [51]:
for row in df2.itertuples():
    if str(row[2]) == 'Docs To Go™ Free Office Suite':
        print("yes")
    elif str(row[2]) == 'Instachat 😜':
        print("yes2")
    elif str(row[2]) == '爱奇艺PPS -《欢乐颂2》电视剧热播':
        print("yes3")


Based on the code above, there are no such applications left in the datset.